### DISCUSSION TITLE
How to get 'Qtree throughtput' info at Harvest?

### DISCUSSION THREAD

**Discussion by wooyoungAhn** [NetApp/harvest#1694] [Q&A] [answered]
Hello there
I hope showing qtree statisics info my harafana dashboard.
How to get 'Qtree throughput info at Harvest?

rest qtree response
{
  "records": [
    {
      "security_style": "unix",
      "nas": {
        "path": "/volume3/qtree1"
      },
      "volume": {
        "_links": {
          "self": {
            "href": "/api/resourcelink"
          }
        },
        "name": "volume1",
        "uuid": "028baa66-41bd-11e9-81d5-00a0986138f7"
      },
      "statistics": {
        "throughput_raw": {
          **"other": 0,
          "read": "200",
          "total": "1000",
          "write": "100"**
        },

query to Prometheus 
**qtree_id**
qtree_id{cluster="CVO", datacenter="AWS", instance="localhost:12993", job="harvest", qtree="qtr1", svm="svm_CVO", volume="nfs_vol"}

**Comment by rahulguptajss** [comment, marked as answer]
@wooyoungAhn Data returned from Rest call `api/storage/qtrees?return_records=true&fields=*,statistics` is raw and still would need to be cooked before it can be used.

**throughput_raw description**:Throughput bytes observed at the interface. This can be used along with delta time to calculate the rate of throughput bytes per unit of time.

Currently Harvest only convert raw performance values to cooked for `ZapiPerf` and `Restperf` collectors. You can enable qtree throughput metrics in Harvest by enabling `Workload` template mentioned [here](https://github.com/NetApp/harvest/blob/main/conf/zapiperf/default.yaml#L58) for `ZapiPerf` Collector.

Below are the Harvest throughput metrics when you enable workload template
https://netapp.github.io/harvest/23.02/ontap-metrics/#qos_read_data
https://netapp.github.io/harvest/23.02/ontap-metrics/#qos_total_data
https://netapp.github.io/harvest/23.02/ontap-metrics/#qos_write_data

Example:
![image](https://user-images.githubusercontent.com/25551691/217513404-66b57994-2936-4580-951f-1ac9526a080a.png)

Qtree qos metrics are available from 9.8+ ONTAP version. More details [here](https://docs.netapp.com/us-en/ontap/performance-admin/guarantee-throughput-qos-task.html#supported-workloads-for-throughput-ceilings)
